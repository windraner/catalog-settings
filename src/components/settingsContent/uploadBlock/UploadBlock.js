import React from 'react';
import UploadLinkBlock from './uploadLinkBlock/UploadLinkBlock';
import UploadBlockSeparator from './uploadBlockSeparator/UploadBlockSeparator';
import UploadFileBlock from './uploadFileBlock/UploadFileBlock';
import PropTypes from 'prop-types';

import './uploadBlock.css';

const UploadBlock = (props) => {
  return (
    <div className="catalog-settings-content__upload-block">
      <UploadLinkBlock
        {...props}
      />

      <UploadBlockSeparator />

      <UploadFileBlock
        {...props}
      />
    </div>
  );
};

UploadBlock.propTypes = {
  inputHandler: PropTypes.func.isRequired,
  fileUrl: PropTypes.string.isRequired,
  file: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.object
  ]).isRequired,
  flags: PropTypes.object,
};

export default UploadBlock;
