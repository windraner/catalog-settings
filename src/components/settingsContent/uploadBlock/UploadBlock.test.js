import React from 'react';
import UploadBlock from './UploadBlock';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json'

configure({ adapter: new Adapter() })

describe('<UploadBlock />', () => {
  it('matches the snapshot', () => {
    const tree = shallow(<UploadBlock
      inputHandler={function() {}}
      fileUrl=""
      file=""
      flags={{}}
    />)
    expect(toJson(tree)).toMatchSnapshot()
  })
});
