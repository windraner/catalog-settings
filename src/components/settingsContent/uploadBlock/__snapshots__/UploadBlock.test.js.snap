// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<UploadBlock /> matches the snapshot 1`] = `
<div
  className="catalog-settings-content__upload-block"
>
  <UploadLinkBlock
    file=""
    fileUrl=""
    flags={Object {}}
    inputHandler={[Function]}
  />
</div>
`;
